### PR TITLE
Feature 5.2 5.5 upgrading renovation and canceling renovation

### PR DESCRIPTION
**Pull request by kmilena98**
Canceling renovation, examination, and equipment transfer and in process of complex renovation, we have added an equipment transfer between rooms that are renovating.

### CODE REVIEW

File: HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs [hunk 1]
```diff
@@ -148,7 +148,16 @@ public ICollection<Examination> GetExaminationsByPatientAndDateTime(int patientC
...
         }
-
+        public void DeleteExaminationRepository(int id)
+        {
+            Examination examination = _context.Examinations.SingleOrDefault(d => d.Id == id);
+            if (examination != null) 
+            {
+                _context.Remove(examination);
+                _context.SaveChanges();
+            }
```
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs]: Why delete the Examination? They can already be cancelled.

File: HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ public void DeleteRoom(int number)
 
         public List<Room> GetAllRooms()
         {
-            return _context.Rooms.ToList();
+            return _context.Rooms.Where(d=> !d.Renovation).ToList();
```
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs:32]: Does this cause issues with other parts of the system that use this getter? You are essentially using the renovation field for logical deletion, that's kind of weird?? If you want to logically delete rooms rename this field.

File: HealthcareSystem/Backend/Model/DataSeeder.cs [hunk 3]
```diff
@@ -83,6 +85,24 @@ private void SeedCountries(MyDbContext context)
...
+            context.Add(new MergeRenovation(new RenovationPeriod(new DateTime(2021, 4, 10, 11, 20, 0, DateTimeKind.Utc), new DateTime(2021, 4, 11, 12, 45, 0, DateTimeKind.Utc)),"plafonjera",TypeOfRenovation.MERGE_RENOVATION,56,57,"soba za konsultacija",TypeOfUsage.CONSULTING_ROOM));
+
+            context.Add(new DivideRenovation(new RenovationPeriod(new DateTime(2021, 4, 12, 11, 20, 0, DateTimeKind.Utc), new DateTime(2021, 4, 12, 12, 45, 0, DateTimeKind.Utc)), "gletovanje", TypeOfRenovation.MERGE_RENOVATION, 18,"soba za konsultacija","soba za operaciju", TypeOfUsage.CONSULTING_ROOM,TypeOfUsage.OPERATION_ROOM));
+            context.Add(new DivideRenovation(new RenovationPeriod(new DateTime(2021, 4, 9, 11, 20, 0, DateTimeKind.Utc), new DateTime(2021, 4, 10, 12, 45, 0, DateTimeKind.Utc)), "menjanje instalacije", TypeOfRenovation.MERGE_RENOVATION, 19, "soba za konsultacija", "soba za operaciju", TypeOfUsage.CONSULTING_ROOM, TypeOfUsage.OPERATION_ROOM));
+
+            context.SaveChanges();
+        }
```
Comment (milana-todorovic) [inline HealthcareSystem/Backend/Model/DataSeeder.cs:104]: If you want the seeding to be on Heroku you have to add it to SeededDbContext as well.

File: HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs (hunk 2 above)
Comment (kmilena98) [reply HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs:32]: That room will be created after renovation, so we are unable to use it before the room division.

File: HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs (hunk 1 above)
Comment (kmilena98) [reply HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs]: We have changed it.

File: HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs (hunk 2 above)
Comment (milana-todorovic) [reply HealthcareSystem/Backend/Repository/RoomRepository/MySqlRoomRepository/MySqlRoomRepository.cs:32]: Alright.

File: HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs (hunk 1 above)
Comment (milana-todorovic) [reply HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs]: Use the Cancel method that already exists in the ExaminationService class, or preferably the cancel endpoint in the ScheduleService. There is no need to reimplement this functionality on this branch.

Comment (kmilena98) [reply HealthcareSystem/Backend/Repository/ExaminationRepository/MySqlExaminationRepository/MySqlExaminationRepository.cs]: Okay, l have changed it.

**Review by milana-todorovic — APPROVED** [review]
Seems alright, as far as I have checked there are no issues with the rest of the system.

**Review by VasilijeBursac — APPROVED** [review]

**Pull request merged**